packages/pwa/app/utils/url.js

```diff
@@ -185,3 +196,67 @@ export const removeQueryParamsFromPath = (path, keys) => {
 
     return `${pathname}${paramStr && '?'}${paramStr}`
 }
+
+/**
+ * Rebuild the path with locale/site value to the path as url path or url query param
+ * based on url config
+ * @param {string} url - based url of the output url
+ * @param {object} configValues - object that contains values of url param config
+ * @return {string} - an output url
+ *
+ * @example
+ * //pwa-kit-config.json
+ * url {
+ *    locale: "query_param"
+ * }
+ * buildPathWithLocaleAndSiteParams('/women/dresses', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /women/dresses?locale=en-GB
+ *
+ *  @example
+ * //pwa-kit-config.json
+ * url {
+ *    locale: "path"
+ * }
+ * buildPathWithLocaleAndSiteParams('/women/dresses', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /en-GB/women/dresses
+ *
+ */
+export const buildPathWithUrlConfigParams = (url, configValues = {}) => {
```

Comment:
I feel like `buildPathWithUrlConfigParams` is a little bit of a misnomer. We aren't using parameters to build the path, we are using the parameter _values_, but `buildPathWithUrlConfigParamValues` is awfully long. Maybe `buildPathWithUrlConfig` is enough? Or something like `buildQualifiedPath`

Reply:
`buildPathWithUrlConfig` makes sense to me. Thanks for the suggestion. 👍